bugsnag-android-core/src/main/java/com/bugsnag/android/internal/ImmutableConfig.kt

```diff
@@ -181,6 +181,33 @@ internal fun convertToImmutableConfig(
     )
 }
 
+private fun validateApiKey(value: String) {
+    if (isInvalidApiKey(value)) {
+        DebugLogger.w(
+            "Invalid configuration. apiKey should be a 32-character hexademical string, got $value"
+        )
+    }
+}
+
+@VisibleForTesting
+fun isInvalidApiKey(apiKey: String): Boolean {
+    if (apiKey.isNullOrEmpty()) {
+        throw IllegalArgumentException("No Bugsnag API Key set")
+    }
+    if (apiKey.length != VALID_API_KEY_LEN) {
+        return true
+    }
+    // check whether each character is hexadecimal (either a digit or a-f).
+    // this avoids using a regex to improve startup performance.
+    for (k in 0 until VALID_API_KEY_LEN) {
+        val chr = apiKey[k]
+        if (!chr.isDigit() && chr !in 'a'..'f') {
+            return true
+        }
+    }
+    return false
```

Comment:
Can't believe I didn't see this last time around 🙈 

```suggestion
    return !apiKey.all { it.isDigit() || it in 'a'..'f' }
```